File: app/src/main/java/ie/dublinbuspal/android/view/nearby/NearbyView.java
package ie.dublinbuspal.android.view.nearby;

import com.hannesdorfmann.mosby3.mvp.MvpView;

import java.util.List;

import ie.dublinbuspal.model.stop.Stop;
import ie.dublinbuspal.util.Coordinate;

public interface NearbyView extends MvpView {

    void launchRealTimeActivity(String stopId);

    void showNearbyStops(List<Stop> busStops);

    void hideProgress();

    void showButton();

    void updateLocation(Coordinate coordinate);

}

File: app/src/main/java/ie/dublinbuspal/android/view/nearby/NearbyView.kt
package ie.dublinbuspal.android.view.nearby

import com.hannesdorfmann.mosby3.mvp.MvpView
import ie.dublinbuspal.domain.model.stop.Stop
import java.util.SortedMap

interface NearbyView : MvpView {

    fun showBusStops(stops: SortedMap<Double, Stop>)

}
